repositories/strategies/routingStrategy.js

```diff
@@ -183,37 +189,47 @@ const toggleConditionValueStrategy = async (
 async function getAvailableRoutingDestinations(trx, pageId) {
   const page = await getPageById(trx, pageId);
   const section = await getSectionById(trx, page.sectionId);
+  const questionPages = await getPageDestinations(trx, page);
+  const sections = await getSectionDestinations(trx, section);
 
-  const results = await Promise.all([
-    getPageDestinations(trx, page),
-    getSectionDestinations(trx, section)
-  ]);
-
-  return flatten(results);
+  return {
+    questionPages,
+    sections
+  };
 }
 
-function checkRoutingDestinations(
+const checkRoutingDestinations = async (
   availableRoutingDestinations,
-  sectionDestination,
-  pageDestination
-) {
-  const [questionPages, sections] = partition(
-    availableRoutingDestinations,
-    "sectionId"
-  );
+  destination
+) => {
+  const { logicalDestinations } = availableRoutingDestinations;
+  const { logicalDestination, absoluteDestination } = destination;
 
-  if (isNil(pageDestination)) {
-    if (!find(sections, { id: parseInt(sectionDestination) })) {
+  if (!isNil(logicalDestination)) {
+    if (
+      !find(logicalDestinations, {
+        logicalDestination: logicalDestination.destinationType
+      })
+    ) {
       throw new Error(
-        `Unable to route from this question to section ${sectionDestination}`
+        `Unable to route from this question ${logicalDestination.destinationType}`
```

Comment:
Could you do this on with one if statement instead by using 
`if(includes(logicalDestinations, logicalDestination.destinationType))`

Reply:
`includes` won't work here since it matches only on value whereas `find` allows you to specify a predicate function or identity.